packages/grid/x-data-grid/src/hooks/features/filter/gridFilterUtils.ts

```diff
@@ -239,18 +239,24 @@ export const buildAggregatedQuickFilterApplier = (
   );
 
   return (rowId, shouldApplyFilter) => {
+    const rowNode = apiRef.current.getRowNode(rowId);
+    if (rowNode?.isAutoGenerated) {
+      return true;
+    }
```

Comment:
The row grouping filter is expecting that autogenerated rows pass the filtering.

For normal filtering, it makes sense. If you set "director must contain David" and you groupes your table with "director" you don't want rows which are not a child of a director named "David"


For the quick filter, the logic is a bit different since the criterion is applied without specifying the logic.
So for the following structure
- david
	- 1
	- 2
	
The quick search "1" will return nothing because the grouping row "david" does not contain "1"
The quick search "David" will return nothing because the grouping row "david" does not contain row passing the quick filter

I made a fix saying that all grouping rows pass quick filtering assuming the user will quick search in the child more often than the grouping columns

Reply:
What should be the behavior on Tree Data ?
You have different filtering logic for Row Grouping and for Tree Data

Reply:
From what I understood, with tree data we only filter leaf rows and auto-generate rows to show the tree structure

So letting autogenerated rows pass the filter should not ba a problem

Reply:
I think we could end up with all the children of an auto generated group filtered-out and the group still visible
But I'm not sure

Reply:
I confirm it lets group without leafs 

https://codesandbox.io/s/treedatasimple-demo-mui-x-forked-3u7zey?file=/demo.tsx

Reply:
I added a test to be sure not breaking the behavior the same way for tree

For the initial problem on row grouping, I just reverted the modification. It should be done in another PR. For now I don't know how to do it